dev/fattest.simplicity/src/componenttest/topology/database/container/LibertyClientUtil.java

```diff
@@ -0,0 +1,194 @@
+/*******************************************************************************
+ * Copyright (c) 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package componenttest.topology.database.container;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.testcontainers.containers.JdbcDatabaseContainer;
+
+import com.ibm.websphere.simplicity.config.ClientConfiguration;
+import com.ibm.websphere.simplicity.config.ConfigElementList;
+import com.ibm.websphere.simplicity.config.DataSource;
+import com.ibm.websphere.simplicity.config.DataSourceProperties;
+import com.ibm.websphere.simplicity.config.DatabaseStore;
+import com.ibm.websphere.simplicity.config.Variable;
+import com.ibm.websphere.simplicity.config.dsprops.Properties;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.topology.impl.LibertyClient;
+
+/**
+ * Client specific liberty config utilities used by the fattest.database project
+ */
+public class LibertyClientUtil {
+
+    //Logging Constants
+    private static final Class<LibertyClientUtil> c = LibertyClientUtil.class;
+
+    //No objects should be created from this class
+    private LibertyClientUtil() {}
+
+    /**
+     * For use when attempting to use <b>database rotation</b>. <br>
+     *
+     * Retrieves database specific properties from the provided JdbcDatabaseContainer, such as;
+     * username, password, etc. <br>
+     *
+     * Using the ClientConfiguration API. Retrieves all &lt;dataSource&gt; elements and modifies
+     * those that have the <b>fat.modify=true</b> attribute set. <br>
+     *
+     * This will replace the datasource &lt;properties.derby.*... &gt; with the specific &lt;properties.db... &gt
+     * for the provided JdbcDatabaseContainer. <br>
+     *
+     * @see              com.ibm.websphere.simplicity.config.ClientConfiguration
+     *
+     * @param  client    - LibertyClient client instance being used for this FAT suite.
+     * @param  cont      - JdbcDatabaseContainer instance being used for database connectivity.
+     *
+     * @throws Exception
+     */
+    public static void setupDatabaseProperties(LibertyClient client, JdbcDatabaseContainer<?> cont) throws CloneNotSupportedException, Exception {
```

Comment:
Any benefit to keeping the `setupDatabaseProperties` and `setupGenericProperties` methods separate? The only difference I noticed was in the last argument for the modify method (true/false).

Reply:
You are correct, I'll fix that.